src/_pytest/config/__init__.py

```diff
@@ -152,6 +153,8 @@ def main(
     :returns: An exit code.
     """
     try:
+        old_pytest_version = os.environ.get("PYTEST_VERSION")
```

Comment:
Very minor as here it cannot happen, but good practice is to define the variable you will be using in `finally` outside the `try` statement, in case the code that defines the variable raises an exception for some reason, which would leave the variable undefined and cause an `UnboundLocalError` to happen in `finally`.

As I said impossible in this case, but it does not cost anything to follow the good practice in this case. :+1: 

```suggestion
    old_pytest_version = os.environ.get("PYTEST_VERSION")
    try:
```